### PR TITLE
hashlink: add haxelib library to output

### PR DESCRIPTION
**Pull request by MadMcCrow**
## Description of changes

This adds an output folder to `pkgs.hashlink`. hashlink haxelib library (located in `other/haxelib`) is necessary to use the HL/C pipeline, allowing native compilation (in C with GCC) of haxe code. The library is then detected by haxe (if in `lib/haxe/{name}/{versionWithCommas}/`) 
details about hashlink compilation can be found in the [haxe documentation](https://haxe.org/manual/target-hl-c-compilation.html)

Another way would be to do a separate package in haxePackages. but since it uses the same source I don't know if it's sane to not re-use.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/hashlink/default.nix [hunk 1]
```diff
@@ -44,6 +44,17 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ getconf ];
 
+  # append default installPhase with library install for haxe
+  postInstall = let
+    haxelibPath =
+      "$out/lib/haxe/${pname}/${lib.replaceStrings [ "." ] [ "," ] version}";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/hashlink/default.nix:50]:
```suggestion
      "$out/lib/haxe/hashlink/${lib.replaceStrings [ "." ] [ "," ] version}";
```

**Comment by duvetfall** [comment]
Could this be merged, please? 
I'm able to build native executable with this PR.

```console
$ cat Main.hx
class Main {
    static public function main():Void {
        trace("Hello World");
    }
}

$ nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/pull/257628/head.tar.gz \
  -p haxe hashlink --pure --run '
    mkdir -p out &&
    haxe -hl out/main.c -main Main.hx &&
    gcc -O3 -o out/main -std=c11 -lm -I out out/main.c -lhl
  '
Code generated in out/main.c automatic native compilation not yet implemented

$ ./out/main 
Main.hx:3: Hello World
```

**Comment by MadMcCrow** [comment]
I don't know what I can do to get this merged, but surely, there's a maintainer that can look at it and merge it?

**Comment by AndersonTorres** [comment]
Just to make sure, can you run the formatter `nixfmt-rfc-style` on this code?

**Comment by AndersonTorres** [comment]
If anything, I will post this PR on Discourse and wait for anyone interested in merging this.

https://discourse.nixos.org/t/prs-ready-for-review/3032/

File: pkgs/development/interpreters/hashlink/default.nix [hunk 2]
```diff
@@ -44,6 +44,17 @@ stdenv.mkDerivation rec {
...
+    haxelibPath =
+      "$out/lib/haxe/hashlink/${lib.replaceStrings [ "." ] [ "," ] version}";
+  in ''
+    mkdir -p "${haxelibPath}"
+    echo -n "${version}" > "${haxelibPath}/../.current"
+    cp -r other/haxelib/* "${haxelibPath}"
+  '';
+
+
   postFixup = lib.optionalString stdenv.isDarwin ''
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hashlink/default.nix:58]:
```suggestion

  postFixup = lib.optionalString stdenv.isDarwin ''
```

File: pkgs/development/interpreters/hashlink/default.nix [hunk 3]
```diff
@@ -44,6 +44,17 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ getconf ];
 
+  # append default installPhase with library install for haxe
+  postInstall = let
+    haxelibPath =
+      "$out/lib/haxe/hashlink/${lib.replaceStrings [ "." ] [ "," ] version}";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hashlink/default.nix:50]:
```suggestion
    haxelibPath = "$out/lib/haxe/hashlink/${lib.replaceStrings [ "." ] [ "," ] version}";
```

**Comment by SuperSandro2000** [comment]
@ofborg build hashlink

**Pull request merged**